### ISSUE TITLE
yolo26 models local reimplement on coco val, can not reach the same mAP from readme.

### ISSUE THREAD

**Issue by wufan-tb** [ultralytics/ultralytics#24624]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I tried the command "yolo val detect model=yolo26l.pt data=coco.yaml device=0", yet only got 0.535 mAP50-95, less than 0.550(cite from official docs); 

here are some details may lead to this situation:
- the coco val data I downloaded, has 48 background images(no annotation), I am not sure is that normal;
- when convert coco format to yolo format, I keep all bbox including iscrowded, should it be droped?

so, why can't I reimplement the mAP value, any help would be appreciated.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wufan-tb, thank you for your interest in Ultralytics 🚀! This is an automated response to help gather the right details, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.
If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).
For your validation reproducibility question, it would be especially helpful if you could share:
-Your exact `yolo val` command
-Your `ultralytics` package version
-Any relevant validation logs or screenshots
-Details on the COCO download/source and any preprocessing or format conversion steps you applied
-A small [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) if you believe this may be a bug
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by lmycross** [comment]
0.55 mAP is achieved in non-end2end mode, you can get this result by running:
```
yolo val detect model=yolo26l.pt data=coco.yaml device=0 end2end=False
```
And the "iscrowded" data should be dropped. When you run the code above, the YOLO format COCO dataset will be downloaded automatically if it is not found locally.

**Comment by wufan-tb** [comment]
> 0.55 mAP is achieved in non-end2end mode, you can get this result by running:
> 
> ```
> yolo val detect model=yolo26l.pt data=coco.yaml device=0 end2end=False
> ```
> 
> And the "iscrowded" data should be dropped. When you run the code above, the YOLO format COCO dataset will be downloaded automatically if it is not found locally.

Hi，thanks for your reply.  I droped the "iscrowded" data and set end2end=False, the value comes to 0.545, perhaps there are other details that I haven't noticed？

here are my command: `yolo val detect model=pretrain_weights/yolo26l.pt data=configs/data/coco.yaml device=0 end2end=False`
and the log report:

<img width="2319" height="350" alt="Image" src="https://github.com/user-attachments/assets/9d7227bc-19e4-4045-aebf-93fb2f8cdef0" />

**Comment by lmycross** [comment]
Maybe the gap is caused by the data, you can try to convert the data using the ultralytics conversion function:
```
from ultralytics.data.converter import convert_coco
convert_coco(labels_dir="path/to/coco/annotations/", save_dir="path/to/coco_converted")
```

**Comment by wufan-tb** [comment]
> Maybe the gap is caused by the data, you can try to convert the data using the ultralytics conversion function:
> 
> ```
> from ultralytics.data.converter import convert_coco
> convert_coco(labels_dir="path/to/coco/annotations/", save_dir="path/to/coco_converted")
> ```

Hi, I re-convert the dataset by this function, yet the metric still be 0.545.

**Comment by lmycross** [comment]
During validation, results are first evaluated with the internal evaluator and then with `faster-coco-eval`. The internal evaluator is faster, but its mAP may differ slightly from the official coco mAP. Please check the `faster-coco-eval` results for the final coco mAP:
<img width="1228" height="382" alt="Image" src="https://github.com/user-attachments/assets/6a1ced79-a2da-44c1-ab16-e625779dd26f" />

**Comment by wufan-tb** [comment]
How can I get the faster-coco-eval results, I don't find the predictions.json

**Comment by wufan-tb** [comment]
I save the json results, and then eval with faster-coco-eval, yet a very low result appeard, maybe it casued by bbox label not matched.(yolo has continuous 80 categories, and coco are not).

anyway, if your local results with command `yolo detect val` is 0.545, I think my local results are matched, thanks for your help.

<img width="1007" height="505" alt="Image" src="https://github.com/user-attachments/assets/40b236d0-194d-4eae-910a-cf2548bec9ec" />

**Comment by glenn-jocher** [comment]
Yes, that’s likely the cause: COCO JSON uses non-contiguous category IDs, while YOLO labels are `0-79`, and Ultralytics handles that mapping automatically when writing `predictions.json` in `pred_to_json()` and running COCO eval in `eval_json()` as shown in the [validator reference](https://docs.ultralytics.com/reference/models/yolo/detect/val) and [Val docs](https://docs.ultralytics.com/modes/val); if your `yolo val ... end2end=False` result is ~`0.545`, your reproduction looks aligned.
